MD/md-3/asychronous-upgrades-problem.md

```diff
@@ -0,0 +1,73 @@
+Consider two honest partitions of the network A and B. At a given block height $h$, Partition A upgrades to a new version of the chain containing a state derivation $S_A(h)$. At the same block height, Partition B remains on the old version of the chain containing a state derivation $S_B(h)$ s.t. $S_A(h) \neq S_B(h)$.
+
+Because a fixed proportion of the stake is needed to agree on the current block height $h$, the network will not be able to agree on $h$. 
+
+The network may be able to provisionally agree on $h\' > h$ within the block tolerance $\tau$ of the network. However, these agreements are never realized because $h$ remains undecided. 
+
+## Solving the Problem with Partially Synchronous Forks
+The Asynchronous Upgrades Problem can be resolved by allowing the network to fork in a partially synchronous manner. That is, the network can agree at a defined point in time $t_{h - 1} + \epsilon$ to fork into two chains w.l.o.g., one with state derivation $S_A(h)$ and the other with state derivation $S_B(h)$.
+
+Within the current epoch, stake weights would then be split between the two chains, Fork A and Fork B, w.l.o.g. based on the votes represented at the fork point at the time of splitting. In other words, the votes are deemed unanimous at the fork point.
+
+In the literal case of an upgrade, the Partition B may then rejoin its stake in Partition A in the next epoch. This effectively means, however, that Partition B would not have any stake-based incentive to complete its upgrade, instead preferring to continue to commit to the fork it created until the next epoch.
+
+In the event a long-lived fork is intended by Partition B, MCR will have provided a usable means to facilitate this fork. 
+
+## Fork Creation Problem
+Assume towards definition of the problem that rewards issued on Fork B are non-zero in perpetuity and that slashing only occurs at the fork point owing to the initial disagreement. 
+
+**Rationale**: for a continued Fork B to be rational for Partition B, we state that 
+
+```math
+U_B = U_{B, s} + \frac{U_{B, r}}{1 - \gamma} + U_{B, p} \geq U_{A, s} + \frac{U_{A, r}}{1 - \gamma} = U_A 
+```
+
+That is, the utility of the state represented by Fork B $U_{B, s}$ plus the discounted repeated utility of the rewards $\frac{U_{B, r}}{1 - \gamma}$ plus some penalty $U_{B, p} < 0$ must be greater than or equal to the utility of the state represented by Fork A $U_{A, s}$ plus the discounted repeated utility of the rewards $\frac{U_{A, r}}{1 - \gamma}$. (Note the utility of the state could also be assigned a discounting model, however, this is considered extrinsic to the model of the problem provided herein.)
+
+**Positive Rewards for Fork B**: since rewards on Fork B are non-zero and Partition B is in agreement with Fork B, it follows that $\frac{U_{B, r}}{1 - \gamma} > 0$
+
+**Initial Penalty for Fork B**: because there was an initial disagreement at the fork point leading to a slash of Partition B's stake, it follows that $U_{B, p} < 0$
```

Comment:
```suggestion
**Initial Penalty for Fork B**: because there was an initial disagreement at the fork point leading to a slashing of Partition B's stake, it follows that $U_{B, p} < 0$
```